ls:  
  lslidar_driver_node:
    ros__parameters:
      lidar_type: "LS"            # 雷达类型
      lidar_model: "LSS3"         # 雷达型号: LSS3 LSS4
      device_ip: "192.168.1.200"  # 雷达IP地址
      msop_port: 2368             # 雷达目的数据端口
      difop_port: 2369            # 雷达目的设备端口
      packet_rate: 15000.0        # PCAP文件回放速率，离线解析PCAP数据时使用
      add_multicast: false        # 雷达是否开启组播
      group_ip: "224.1.1.2"       # 组播IP地址
      packet_loss: false          # 是否开启丢包检测
      #pcap: "/home/ls/Desktop/lidar_data.pcap"          # PCAP文件路径，离线解析PCAP数据时打开注释
      
      # 点云处理参数
      is_add_frame: false         # 是否叠帧发布点云信息
      frame_id: "laser_link"      # 坐标系名称
      pointcloud_topic: "lslidar_point_cloud"  # 点云话题名
      use_time_service: false     # 雷达是否使用授时(GPS PTP NTP)
      use_first_point_time: false # true: 使用每帧第一个点的时间做为点云时间
      use_absolute_time: false    # 点时间格式 true: 使用绝对时间  注意与点云头文件中USE_ABSOLUTE_TIME保持一致
      min_range: 0.15             # 雷达扫描最小距离 小于该值的点将被过滤(m)
      max_range: 2000.0           # 雷达扫描最大距离 大于该值的点将被过滤(m)
      scan_start_angle: -6000     # 雷达扫描最小角度 填整数 单位: 0.01°
      scan_end_angle: 6000        # 雷达扫描最大角度 填整数 单位: 0.01°

      # 点云坐标系转换参数
      is_pretreatment: false      # 预处理开关 
      x_offset: 0.0               # X轴平移(m)
      y_offset: 0.0               # Y轴平移(m)
      z_offset: 0.0               # Z轴平移(m)
      roll: 0.0                   # X轴旋转(rad)
      pitch: 0.0                  # Y轴旋转(rad)
      yaw: 0.0                    # Z轴旋转(rad)

      is_MatrixTransformation: false  # 预处理开关 true 使用以下两个(4*4)矩阵进行转换(若只转换一次，可将其中一个删除)
      transform_main:
        [
          0.3514713644981384,
          0.3227383494377136,
          0.878810465335846,
          1.250789999961853,
          -0.17116132378578186,
          0.9450311660766602,
          -0.27860331535339355,
          -0.3601331412792206,
          -0.9204192757606506,
          -0.05249723792076111,
          0.38739171624183655,
          1.9063990116119385,
          0.0,
          0.0,
          0.0,
          1.0,
        ]
      transform_imu:
        [
          0.999537,
          -0.030429,
          0.00027,
          -1.047623,
          0.030413,
          0.999231,
          0.024782,
          5.966497,
          -0.001024,
          -0.024763,
          0.999693,
          0.638128,
          0.0,
          0.0,
          0.0,
          1.0,
        ]

